bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/service/SonyServlet.java

```diff
@@ -0,0 +1,208 @@
+            logger.debug("Started Sony Web service at {}", SONY_PATH);
+        } catch (ServletException | NamespaceException e) {
+            logger.debug("Exception starting status servlet: {}", e.getMessage(), e);
+        }
+    }
+
+    @Deactivate
+    public void deactivate() {
+    }
+
+    @Override
+    protected void doPost(final @Nullable HttpServletRequest req, final @Nullable HttpServletResponse resp)
+            throws ServletException, IOException {
+        Objects.requireNonNull(req, "req cannot be null");
+        Objects.requireNonNull(resp, "resp cannot be null");
+        final CommandRequest cmdRqst = gson.fromJson(req.getReader(), CommandRequest.class);
+
+        final String baseUrl = cmdRqst.getBaseUrl();
+        if (baseUrl == null || StringUtils.isEmpty(baseUrl)) {
+            write(resp, gson.toJson(new CommandResponse(false, "baseUrl is required")));
+            return;
+        }
+
+        final String serviceName = cmdRqst.getServiceName();
+        if (serviceName == null || StringUtils.isEmpty(serviceName)) {
+            write(resp, gson.toJson(new CommandResponse(false, "serviceName is required")));
+            return;
+        }
+
+        final String transportName = cmdRqst.getTransport();
+        if (transportName == null || StringUtils.isEmpty(transportName)) {
+            write(resp, gson.toJson(new CommandResponse(false, "transport is required")));
+            return;
+        }
+
+        final String command = cmdRqst.getCommand();
+        if (command == null || StringUtils.isEmpty(command)) {
+            write(resp, gson.toJson(new CommandResponse(false, "command is required")));
+            return;
+        }
+
+        final String version = cmdRqst.getVersion();
+        if (version == null || StringUtils.isEmpty(version)) {
+            write(resp, gson.toJson(new CommandResponse(false, "version is required")));
+            return;
+        }
+
+        final String parms = cmdRqst.getParms();
+        // cannot be null but can be empty
+        if (parms == null) {
+            write(resp, gson.toJson(new CommandResponse(false, "parms is required")));
+            return;
+        }
+
+        final SonyTransportFactory factory = new SonyTransportFactory(new URL(baseUrl), gson, webSocketClient,
+                scheduler);
+
+        try (final SonyTransport transport = factory.getSonyTransport(serviceName, transportName)) {
+            if (transport == null) {
+                write(resp, gson.toJson(new CommandResponse(false, "No transport of type: " + transportName)));
+                return;
+            } else {
+                // Use 1 to not conflict with ScalarWebRequest IDs
+                final String cmd = "{\"id\":1,\"method\":\"" + command + "\",\"version\":\"" + version
+                        + "\",\"params\":[" + parms + "]}";
+
+                final ScalarWebRequest rqst = gson.fromJson(cmd, ScalarWebRequest.class);
```

Comment:
lol - I'd love to but then I'd have to parse the parms (which are a simple string).  This was the 'easy' and 'safer' way of doing it.  So basically, I can either json the parms then set the constructor or simply json the whole class.  

Since this serves the webpage and isn't entirely critical for anything - I took the easier way